src/finetining/NLLB/eval.py

```diff
@@ -0,0 +1,161 @@
+import nltk
+import unidecode
+import json
+import sys
+import re
+import string
+import gc
+import torch
+
+
+from nltk.translate import bleu_score
+from typing import List
+from transformers import NllbTokenizer, AutoModelForSeq2SeqLM
+from tqdm import trange
+from argparse import ArgumentParser
+
+nltk.download('punkt')
+
+
+lang_code = {
+
+    "yor": "yor_Latn",
+    "hau": "hau_Latn",
+     "wolof": "wol_Latn",
+     "fon": "swa_Latn",
+     "en": "eng_Latn"
+
+}
+
+
+def cleanup():
+    """Try to free GPU memory"""
+    gc.collect()
+    torch.cuda.empty_cache()
+
+
+def load_model(model_name):
+    model = AutoModelForSeq2SeqLM.from_pretrained(model_name)
+    tokenizer = NllbTokenizer.from_pretrained(model_name)
+    return model, tokenizer
+
+
+def translate(model, tokenizer, text, src_lang, tgt_lang, a=32, b=3 ):
+
+    #model, tokenizer = load_model(model_name)
+    tokenizer.src_lang = src_lang
+    tokenizer.tgt_lang = tgt_lang
+
+    inputs = tokenizer(text, return_tensors="pt", padding=True, truncation=True, max_length=1024)
+    model.eval()
+    translation = model.generate(**inputs.to(model.device), forced_bos_token_id = tokenizer.convert_tokens_to_ids(tgt_lang), max_new_tokens = int(a + b * inputs.input_ids.shape[1]), num_beams = 4)
+    return tokenizer.batch_decode(translation, skip_special_tokens = True)
+    
+
+
+def batched(model_name, dataset_path: string, source_lang, target_lang, batch_size = 16):
+
+    model, tokenizer = load_model(model_name)
+    data = process_file(dataset_path=dataset_path)
+    sorted_data = sorted(data, key= lambda item: len(item["translation"][source_lang]), reverse=True)
+
+    result = []
+    for i in trange(0, len(sorted_data), batch_size):
+
+        sorted_batch = sorted_data[i: i + batch_size]
+        batch = [item["translation"][source_lang] for item in sorted_batch]
```

Comment:
this looks good. we might be using the tsv format on gitbub too. I will drop the finetuning script for you to see soon.

Reply:
Okay